ethToElrond/v2/ethToElrondBridgeExecutor_test.go

```diff
@@ -0,0 +1,601 @@
+		expectedErr := errors.New("expected error")
+		args.EthereumClient = &bridgeV2.EthereumClientStub{
+			GetBatchCalled: func(ctx context.Context, nonce uint64) (*clients.TransferBatch, error) {
+				assert.Equal(t, providedNonce, nonce)
+				return nil, expectedErr
+			},
+		}
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+		err := executor.GetAndStoreBatchFromEthereum(context.Background(), providedNonce)
+
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		providedNonce := uint64(8346)
+		expectedBatch := &clients.TransferBatch{}
+		args.EthereumClient = &bridgeV2.EthereumClientStub{
+			GetBatchCalled: func(ctx context.Context, nonce uint64) (*clients.TransferBatch, error) {
+				assert.Equal(t, providedNonce, nonce)
+				return expectedBatch, nil
+			},
+		}
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+		err := executor.GetAndStoreBatchFromEthereum(context.Background(), providedNonce)
+
+		assert.Nil(t, err)
+		assert.True(t, expectedBatch == executor.GetStoredBatch()) // pointer testing
+	})
+}
+
+func TestEthToElrondBridgeExecutor_GetLastExecutedEthBatchIDFromElrond(t *testing.T) {
+	t.Parallel()
+
+	args := createMockEthToElrondExecutorArgs()
+	providedBatchID := uint64(36727)
+	args.ElrondClient = &bridgeV2.ElrondClientStub{
+		GetLastExecutedEthBatchIDCalled: func(ctx context.Context) (uint64, error) {
+			return providedBatchID, nil
+		},
+	}
+	executor, _ := NewEthToElrondBridgeExecutor(args)
+
+	batchID, err := executor.GetLastExecutedEthBatchIDFromElrond(context.Background())
+	assert.Equal(t, providedBatchID, batchID)
+	assert.Nil(t, err)
+}
+
+func TestEthToElrondBridgeExecutor_VerifyLastDepositNonceExecutedOnEthereumBatch(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil batch should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+
+		err := executor.VerifyLastDepositNonceExecutedOnEthereumBatch(context.Background())
+		assert.Equal(t, errNilBatch, err)
+	})
+	t.Run("get last executed tx id errors", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		expectedErr := errors.New("expected error")
+		args.ElrondClient = &bridgeV2.ElrondClientStub{
+			GetLastExecutedEthTxIDCalled: func(ctx context.Context) (uint64, error) {
+				return 0, expectedErr
+			},
+		}
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+		executor.batch = &clients.TransferBatch{}
+
+		err := executor.VerifyLastDepositNonceExecutedOnEthereumBatch(context.Background())
+		assert.Equal(t, expectedErr, err)
+	})
+
+	args := createMockEthToElrondExecutorArgs()
+	txId := uint64(6657)
+	args.ElrondClient = &bridgeV2.ElrondClientStub{
+		GetLastExecutedEthTxIDCalled: func(ctx context.Context) (uint64, error) {
+			return txId, nil
+		},
+	}
+
+	t.Run("first deposit nonce equals should error", func(t *testing.T) {
```

Comment:
If we are checking here(and in some tests below) only the first deposit/entry in deposits slice (which should be wrong => expect error => expect stop looping through other deposits), is it worth adding a second entry in slice? E.g., here: `Nonce: txId + 2`.
Also, a small naming suggestion here and below:
```suggestion
	t.Run("first deposit nonce equals last tx nonce, should error", func(t *testing.T) {
```

Reply:
right, I've copy-pasted the code. Removed the second deposit transfer
Renamed the tests